plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/async/AsyncExecutorBuilder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.s3.async;
+
+import java.util.concurrent.ExecutorService;
+
+/**
+ * An encapsulation for the {@link TransferNIOGroup}, and the stream reader and future completion executor services
+ */
+public class AsyncExecutorBuilder {
+
+    private final ExecutorService futureCompletionExecutor;
+    private final ExecutorService streamReader;
```

Comment:
Why should stream reader(specific operation) be a part of the generic executor.

Reply:
Didn't get this